src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -162,20 +162,23 @@ public void Dispose()
             {
                 throw new ArgumentNullException("role");
             }
-
             var result = await ValidateRoleInternal(role);
             if (!result.Succeeded)
             {
                 return result;
             }
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(await Store.CreateAsync(role, CancellationToken), role);
+            result = await Store.CreateAsync(role, CancellationToken);
+            using (IdentityLogger.BeginScope(await GetScopeMessageforRoleAsync(role)))
+            {
+                return IdentityLogger.Log(result);
```

Comment:
@divega i can see the point. This behavior is only for CreateAsync on users and roles since we wouldn't have access to the id values unless the object is created, especially for store generated ids. We still need to know that the user/role was created though, so we need to log it. Now we can special case this and log only, but need to call the Log method directly that takes result, Func<LogLevel>, Func<string> since we need to set the userId in the message instead of scope. I am fine either way. This approach only makes it looks clean